File: leet/larger_in_array.rb
def larger(nums)
    output = 0
    counter = 0
    while counter < nums.length
        first = nums[counter]
        counter_two = counter + 1
        while counter_two < nums.length
            second = nums[counter_two]

            if first < second && ((second - first) > output)
                output = second - first
            end
            counter_two += 1
        end

        counter += 1
    end
    puts output
end

larger([2,3,10,6,4,8,1])
larger([4,3,1])

File: python/larger_in_array.py
def funcMaxDifference(inputArr):
	# Write your code here
	output = 0
	counter = 0
	while counter < len(inputArr):
		first = inputArr[counter]
		counter_two = counter + 1
		while counter_two < len(inputArr):
			second = inputArr[counter_two]
			if first < second and ((second - first) > output):
				output = second - first

			counter_two += 1

		counter += 1

	print(output)

funcMaxDifference([2,3,10,6,4,8,1])